core/templates/dev/head/domain/utilities/ProbabilityService.js

```diff
@@ -0,0 +1,27 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for performing probabilistic operations.
+ */
+
+oppia.factory('ProbabilityService', [function() {
+  return {
+    isProbable: function(ratio) {
```

Comment:
I'm assuming `ratio` is in the range `[0, 1]`? If so, this can be simplified to:

```javascript
return Math.random() < ratio;
```

Although I'm not sure this is the best way to go. @seanlip, what do you think of this service?

Reply:
@brianrodri I made this a service because we might be making different methods that rely on probability, and they could reside here.

Reply:
I think it makes more sense to mock `Math.random` directly, like here: https://commandercoriander.net/blog/2014/03/02/testing-around-random-numbers-in-javascript/

But I'm not sure that's a good idea either. Let's wait for Sean's input.

Reply:
I'm not keen on ProbabilityService. I don't really know what it's supposed to do (its definition is not clear) and I don't think Math.random() actually needs a wrapper anyway since there's no difference between browsers (unlike, say, something like localStorage).

So I would recommend just using Math.random() directly wherever you need to use it.

Reply:
(Unless you have additional use cases that require a ProbabilityService, in which case it would help to mention these.)

Reply:
@seanlip I cant think of any probable additional cases for now. I'll take your suggestion then, and use it explicitly.